content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,154 @@
+
+    {{< note >}}
+
+    To generate certificates and private keys for your cluster using the `openssl` command line tool, see [Certificates (`openssl`)](/docs/concepts/cluster-administration/certificates/#openssl).    
+    You can also use [`cfssl`](/docs/concepts/cluster-administration/certificates/#cfssl).
+
+    {{< /note >}}
+
+1. If your cluster is using bootstrap tokens to join nodes, update the ConfigMap `cluster-info` in the `kube-public` namespace with new CA.
+
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    kubectl get cm/cluster-info --namespace kube-public -o yaml | \
+        /bin/sed "s/\(certificate-authority-data:\).*/\1 ${base64_encoded_ca}" | \
+        kubectl apply -f -
+    ```
+
+1. Update the file against `clientCAFile` in kubelet configuration and `certificate-authority-data` in kubelet.conf to use the new CA on all nodes.
+
+    If your kubelet is not using client certificate rotation update `client-certificate-data` and `client-key-data` in kubelet.conf on all nodes along with the kubelet client certificate file usually found in `/var/lib/kubelet/pki`.
+
+1. Follow these steps on all nodes in a rolling fashion.
+
+    1. Restart kubelet to start using the new CA
+
+    1. Restart any other *[aggregated api servers](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)* or *webhook handlers* to trust the new CA certificates.
+
+    1. Annotate any daemonsets and deployments to trigger new pod creation in a safer rolling fashion.
+
+        Example:
+
+        ```shell
+        for namespace in $(kubectl get namespace -o jsonpath='{.items[*].metadata.name}'); do
+            for name in $(kubectl get deployments -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
+                kubectl patch deployment -n ${namespace} ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
+            done
+            for name in $(kubectl get daemonset -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
+                kubectl patch daemonset -n ${namespace} ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
+            done
+        done
+        ```
+
+1. Once the cluster functionality is successfully verified:
+
+    1. Remove old CA from the files against `--client-ca-file` and `--root-ca-file` flags and restart the control plane components.
+
+    1. Update all service account tokens to include new CA certificate only.
+
+        *    All pods using an in-cluster kubeconfig will eventually need to be restarted to pick up the new SA secret for the old CA to be completely untrusted.
+
+## Alternative approach
+{{< note >}}
+
+Since, the above-mentioned approach is a lot of work, which might not fit everyone's use case. An alternative approach is to replace the cluster.
+
+{{< /note >}}
+
+## Replace the cluster {#replace-the-cluster}
```

Comment:
i'm starting to get convinced that we should not include this in the guide.
after all, the page is called "manual CA rotation" and this solution should be obvious as an alternative.

so please remove it.
other than that leaving final LGTM to SIG Auth!

thank you.

Reply:
@neolit123 removed the approach.